library/objective-c/EnvoyEngineImpl.m

```diff
@@ -10,6 +11,93 @@ static void ios_on_exit() {
   NSLog(@"[Envoy] library is exiting");
 }
 
+typedef struct {
+  __unsafe_unretained EnvoyHTTPFilter *filter;
+} ios_http_filter_context;
+
+// TODO(goaway): refactor and relocate translation
```

Comment:
There's some (egregious) duplication with some of our other mapping code here. It's a little bit tricky to reconcile, because the retain semantics are slightly different. It sure would be nice to though.

Reply:
Sounds good, maybe make this TODO a bit more descriptive so it doesn't lose context over time

Reply:
Isn't it the same as https://github.com/lyft/envoy-mobile/blob/2d86a1c052ec97852ac8161c470649290310fde7/library/objective-c/EnvoyHTTPStreamImpl.m#L21?

Reply:
The bridge semantics are a little different; see my comments below. Happy to talk about it more. My inclination is to leave this as a TODO (with extra description per @rebello95's suggestion, and work on a better solution separately. To provide a little bit more context, I'm considering if we need an actual refcount on our structs. It's kind of a bridge I'd like to avoid crossing if possible, but if we're going to avoid it, I do feel it's important to be strict about our release expectations - which we can't yet reconcile here.

Reply:
Updated TODO. Note that some of these TODOs I'd like to address pretty much immediately after core functionality ships. However, as long as they don't affect the public interfaces that Android and iOS applications adhere to, I'm biasing towards getting that functionality shipped so that the apps can be updated independent of under-the-hood improvements. As always, open to discussing.